File: Arrays/#2191 Sort the jumbled numbers/main.js
function solve(mapping, nums) {
    const mapper = (num) => {
      const newnum = num.toString();
      let tmpnum = "";
      for (let i = 0; i < newnum.length; ++i) {
        tmp = mapping[newnum[i]];
        tmpnum += String(tmp);
      }
      return Number(tmpnum)
    };
  
    const mappedNums = nums.map(num => [mapper(num),num])
    mappedNums.sort((a,b) => a[0]-b[0])
    const res = mappedNums.map(item => item[1]);
    console.log(res)
  }
  
  solve([8, 9, 4, 0, 2, 1, 3, 5, 7, 6], [991, 338, 38]);

File: Arrays/#2191 Sort the jumbled numbers/main.py
def sortJumbled(mapping, nums):
    def map_number(num: int) -> int:
        numstr = str(num)
        mappedstr = "".join(str(mapping[int(digit)]) for digit in numstr)
        return int(mappedstr)

    return sorted(nums, key=lambda num: map_number(num))
